### ISSUE TITLE
C++ consumer->close() doesn't stop reconnecting to broker.

### ISSUE THREAD

**Issue by agriic** [edenhill/librdkafka#678]
# Description

I am using high level consumer. When connection has been droped (all brokers down), i want to stop consumer (without brutal app termination).
After exiting consumer while loop and calling consumer->close(), it tries quite a long time to reconnect to brokers although it shouldn't.

`Used consumer code:

```
    check(conf->set("metadata.broker.list", brokerList, errstr));
    check(conf->set("queue.buffering.max.ms", "1", errstr));
    check(conf->set("fetch.wait.max.ms", "20", errstr));
    check(conf->set("socket.timeout.ms", "20", errstr));
    check(conf->set("group.id", "test_consumer", errstr));
    check(conf->set("rebalance_cb", dynamic_cast<RdKafka::RebalanceCb*>(this), errstr));
    check(conf->set("event_cb", dynamic_cast<RdKafka::EventCb*>(this), errstr));

    consumer = std::shared_ptr<RdKafka::KafkaConsumer>(RdKafka::KafkaConsumer::create(conf.get(), errstr));
    if (!consumer){ LOG << "Failed to create consumer: " << errstr << endl; return; }

    RdKafka::ErrorCode err = consumer->subscribe({topicName});
    if (err) { LOG << "Failed to subscribe to topic: " << RdKafka::err2str(err) << std::endl; return; }

    while (!stop)
    {
        std::shared_ptr<RdKafka::Message> message(consumer->consume(100));
        string msgString;
        switch (message->err())
        {
            case RdKafka::ERR__TIMED_OUT:
            case RdKafka::ERR__PARTITION_EOF:
                break;

            case RdKafka::ERR_NO_ERROR:
                msgString.assign(static_cast<const char *>(message->payload()), message->len());
                LOG << "MSG: " << msgString << endl;
                break;

            case RdKafka::ERR__UNKNOWN_TOPIC:
            case RdKafka::ERR__UNKNOWN_PARTITION:
            case RdKafka::ERR__ALL_BROKERS_DOWN:
            default:
                std::cerr << "Consume failed: " << message->errstr() << std::endl;
                stop = true;
                break;
        }
    }

    cout << " -----  Stopping - Pre close() ------- " << endl;
    consumer->close();
    cout << " -----  Stopping - Post close() ------- " << endl;
    RdKafka::wait_destroyed(100);
```

`
Full consumer code: https://gist.github.com/agriic/a95e6fa2fe5e4ba182212c7e67913fcb

Is there a way to avoid such behaviour or maybe i am doing something wrong?
# How to reproduce
1. Consumer successfully connects to broker
2. Consumer receives couple messages
3. Connection dropes down -> exiting consume loop
4. Calling consumer->close()
# Checklist

Please provide the following information:
- [ ] librdkafka version (release number or git tag): 0.9.1
- [ ] Apache Kafka version: 0.9
- [ ] librdkafka client configuration:
- [ ] Operating system: OSX
- [ ] Using the legacy Consumer
- [X] Using the high-level KafkaConsumer
- [ ] Provide logs (with `debug=..` as necessary) from librdkafka
- [ ] Provide broker log excerpts
- [X] Critical issue

**Comment by incessantier** [comment]
@edenhill @agriic This might be related to what I mentioned in #674.  I also believe I have it fixed in PR #683

**Comment by edenhill** [comment]
If there are offsets to be committed the close() will block for session.timeout.ms and will attempt to reconnect to brokers in the background meanwhile.
